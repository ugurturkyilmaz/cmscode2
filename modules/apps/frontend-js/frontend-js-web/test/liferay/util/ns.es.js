/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ns from '../../../src/main/resources/META-INF/resources/liferay/util/ns.es';

describe('Liferay.Util.ns', () => {
	it('returns an object', () => {
		const namespace = '_ns_';

		const payload = {
			arr1: [1, 2, 3],
			arr2: ['foo', 'bar', 'baz'],
			int1: 1,
			int2: 2,
			obj1: {
				arr3: [1, 2, 3, 'foo', 'bar', 'baz'],
				int3: 3,
				int4: 4,
				string3: 'baz',
			},
			string1: 'foo',
			string2: 'bar',
		};

		const result = ns(namespace, payload);

		expect(result).toBeDefined();
		expect(typeof result).toBe('object');
		expect(result).toMatchSnapshot();
	});
});
