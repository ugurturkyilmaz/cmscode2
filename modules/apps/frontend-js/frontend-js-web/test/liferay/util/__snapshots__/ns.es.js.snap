// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Liferay.Util.ns returns an object 1`] = `
Object {
  "_ns_arr1": Array [
    1,
    2,
    3,
  ],
  "_ns_arr2": Array [
    "foo",
    "bar",
    "baz",
  ],
  "_ns_int1": 1,
  "_ns_int2": 2,
  "_ns_obj1": Object {
    "arr3": Array [
      1,
      2,
      3,
      "foo",
      "bar",
      "baz",
    ],
    "int3": 3,
    "int4": 4,
    "string3": "baz",
  },
  "_ns_string1": "foo",
  "_ns_string2": "bar",
}
`;
